nomt/src/cursor.rs

```diff
@@ -145,10 +168,37 @@ impl PageCacheCursor {
                     };
                     self.pages.prepopulate(page_id);
                 }
+
+                if let (&Some((ref id, _)), SeekMode::RetrieveSiblingLeafChildren, true) = (
+                    &self.cached_page,
+                    seek_mode,
+                    trie::is_leaf(&self.peek_sibling()),
+                ) {
+                    // sibling is a leaf and at the end of the (non-root) page.
+                    // initiate a load of the sibling's page.
+                    let path = last_page_path(&self.path, self.depth);
+                    let page_idx = {
+                        let this_page_idx = path.load_be::<u8>();
+                        if this_page_idx % 2 == 0 {
+                            this_page_idx + 1
+                        } else {
+                            this_page_idx - 1
+                        }
+                    };
+                    let child_page_id = id.child_page_id(page_idx).expect(
+                        "Child index is 6 bits and Pages do not go deeper than the maximum layer, 42"
```

Comment:
Nit: this proof repeats verbatim several times. As discussed, the only big advantage of `expect` over `unwrap` is that it allows to identify the place of a panic even if the line number information mismatches the actual code and the source SHA is not known. Using the same message removes this advantage.